test/built-ins/Promise/all/resolve-not-callable-close.js

```diff
@@ -0,0 +1,47 @@
+// Copyright (C) 2020 the V8 project authors. All rights reserved.
+// This code is governed by the BSD license found in the LICENSE file.
+
+/*---
+description: >
+    Explicit iterator closing if Promise.resolve is not callable
+esid: sec-promise.all
+info: |
+    5. Let result be PerformPromiseAll(iteratorRecord, C, promiseCapability).
+    6. If result is an abrupt completion,
+        a. If iteratorRecord.[[Done]] is false, let result be
+           IteratorClose(iterator, result).
+        b. IfAbruptRejectPromise(result, promiseCapability).
+
+    [...]
+
+    Runtime Semantics: PerformPromiseAll
+
+    [...]
+    5. Let promiseResolve be ? Get(constructor, "resolve").
+    6. If ! IsCallable(promiseResolve) is false, throw a TypeError exception.
+    [...]
+
+flags: [async]
+features: [Symbol.iterator, computed-property-names, arrow-function]
+---*/
+
+let returnCount = 0;
+const iter = { 
+  [Symbol.iterator]: function() {
+    return {
+      return: function() {
+        ++returnCount;
+      }
+    };
+  }
+}
+
+Promise.resolve = "certainly not callable";
+
+Promise.all(iter).then(() => {
+  $DONE('The promise should be rejected, but was resolved');
+}, (reason) => {
+  assert(reason instanceof TypeError);
+}).then($DONE, $DONE);
+
+assert.sameValue(returnCount, 1);
```

Comment:
Duplicate of: https://github.com/tc39/test262/blob/master/test/built-ins/Promise/all/invoke-resolve-get-error-close.js

Reply:
Existing test covers abrupt completion at step 5 of [PerformPromiseAll](https://tc39.es/ecma262/#sec-performpromiseall), while this PR adds coverage for step 6. Same with other methods.

Reply:
Exactly, like @shvaikalesh says. This is covering the case where "resolve" is not callable, while existing tests cover the case that Get(constructor, "resolve") throws.

As an additional data point, we had a situation in V8 where the existing test262 tests were passing but this test failed B-) so it's surely not a duplicate. For more info: https://bugs.chromium.org/p/v8/issues/detail?id=10452